api/client.go

```diff
@@ -535,9 +539,20 @@ func (c *Client) httpPost(content io.ReadSeeker, endpoint, contentType string, r
 
 // APIHostPorts returns a slice of network.HostPort for each API server.
 func (c *Client) APIHostPorts() ([][]network.HostPort, error) {
+	return c.apiHostPortsCall("APIHostPorts")
+}
+
+// APIHostPortsForAgents returns a slice of network.HostPort for each API server,
+// based on the configured management space.
+// If no such space is configured, this will return the same as APIHostPorts().
+func (c *Client) APIHostPortsForAgents() ([][]network.HostPort, error) {
+	return c.apiHostPortsCall("APIHostPortsForAgents")
```

Comment:
I don't think it makes sense to expose APIHostPortsForAgents as the "Client" facade is never meant to be accessed by agents.

Which also means we don't need the refactoring into a helper call.